canmatrix/canmatrix.py

```diff
@@ -404,13 +408,9 @@ def phys2raw(self, value=None):
                 "Value {} is not valid for {}. Min={} and Max={}".format(
                     value, self, self.min, self.max)
                 )
-        raw_value = (value - self.offset) / self.factor
-
-        if not self.is_float:
-            raw_value = int(raw_value)
-        return raw_value
+        return (value - self.offset) / self.factor
 
```

Comment:
If this is supposed to return a value for use on the CAN bus and it's not a floating point signal then the result should be an integer, yes?  Though there could be an argument about what conversion mechanism should be used, truncation vs round vs all the different rounding techniques.  But it should be an integer?

Perhaps there should be two functions.  One for scaling/offset and another that creates an actually bus-usable value?

Side note since this is pre-existing, `phys2raw` returns the physical value per the comment.  Seems backwards.  I would expect 'physical' to be in regular SI/imperial units and 'raw' to be bus values.

Reply:
@altendky 
in dbc factor and offest mean:
`pyhiscalValue = valueInCanFrame * factor + offset;`

`phys2raw` should convert physicalValue to CanValue:
`valueInCanFrame = (pyhiscalValue - offset) / factor` looks right for me?

Is this another 'dbc-default' which should be documented somewhere?

Reply:
@ebroecker

1) the doc string and function names seemed in conflict.

2) this change removes the conversion to an integer.  Unless the signal is floating point on the bus, shouldn't it be made an integer?  Unless we want to switch to having the concept of a bus-scaled value (that is still float, or decimal, etc) separate from an actually bus representable integer.

Reply:
1. I had to read once more...
the function name is right, BUT the doc-str is backward...

2. you are right

Reply:
@pierreluctg

So, all clear on this now I think. You are intending to have this function return a bus-scaled value which is not actually usable on the bus because it still requires an int conversion?  (Truncation, rounding, etc). I don't have any use case for that in mind, but I haven't thought about it much. Generally, it seems like it might be sensible to have though. I'd probably have a different function name though.

Reply:
fixed